torch/distributed/fsdp/_shard_utils.py

```diff
@@ -191,4 +191,10 @@ def _create_chunk_dtensor(
     # HSDP placements: [Replicate(), Shard(0)]
     placements = [Replicate() for _ in range(device_mesh.ndim)]
     placements[-1] = shard_placement  # type: ignore[call-overload]
-    return DTensor.from_local(local_tensor, device_mesh, placements)
+    return DTensor.from_local(
```

Comment:
looking at the fsdp integration code, I think we can separate out two changes:
1. the first change is that I think we should avoid using non-public API from DTensor, specifically `shard_placement._split_tensor` is not public. The way FSDP should do is to not do manual split, but it should do `from_local(replicate()).redistribute(shard(0))`, this should also avoid the bug you seen, as in theory after summon_params, the params are replicated, we don't need to manually split them, DTensor can do this for us already, and the created DTensor should have the correct shape/offset
2. we should still fix `from_local` by passing shape/stride to that from_local can handle the uneven sharding case without communication. But this can be a PR on top of 1

Reply:
Yes, this makes sense. Thanks for the suggestions! @wanchaol 

As suggested, I separated out the changes in FSDP to use public API from DTensor here. https://github.com/pytorch/pytorch/pull/110831 

Will follow up with you regarding the fix for DTensor.from_local().